lib/index.js

```diff
@@ -15,31 +16,65 @@ module.exports = plugin;
 /**
  * Metalsmith plugin to convert markdown files.
  *
- * @param {Object} options (optional)
+ * @param {Object}       options (optional) - options to pass to markdownIt
+ * @param {Object}       options.plugin (optional) - options used by the plugin. Will not be passed to MarkdownIt
+ * @param {string}       options.plugin.pattern - glob pattern for filtering which files to process (passed to minimatch.filter)
+ * @param {string|Array} options.plugin.fields - field or list of fields which MarkdownIt should process
+ * @param {string}       options.plugin.extension - file extension for output files
  * @return {Function}
  */
 
 function plugin(preset, options){
+  var defaults = {
+    pattern: '**/*.@(md|markdown)',
+    fields: 'contents',
+    extension: 'html'
+  }
+  var pluginOptions = defaults;
+
+  // handle cases where a preset isn't specified
+  var pluginOpts = false;
+  if (options && options.plugin) {
+    pluginOpts = options.plugin;
+    delete options.plugin;
+  } else if (!options && typeof preset === 'object' && preset.plugin){
+    pluginOpts = preset.plugin;
+    delete preset.plugin;
+  }
+
+  if (pluginOpts) {
+    // merge defaults with supplied options
+    for (var prop in pluginOpts) { 
+      pluginOptions[prop] = pluginOpts[prop];
+    }
+  }
+  // normalize pluginOptions.fields into an array
+  if (typeof pluginOptions.fields === 'string') pluginOptions.fields = [pluginOptions.fields]
+
   var markdown = new markdownIt(preset, options),
       envSetter = function(){};
 
   var plugin = function(files, metalsmith, done){
     setImmediate(done);
-    Object.keys(files).forEach(function(file){
-      debug('checking file: %s', file);
-      if (!is_markdown(file)) return;
+    Object.keys(files).filter(minimatch.filter(pluginOptions.pattern)).forEach(function(file){
       var data = files[file];
       var dir = dirname(file);
-      var html = basename(file, extname(file)) + '.html';
-      if ('.' != dir) html = join(dir, html);
+      var html = basename(file, extname(file)) + '.' + pluginOptions.extension;
+      if (dir !== '.') html = join(dir, html);
 
-      debug('converting file: %s', file);
       var env = {};
       if (envSetter) {
         env = envSetter(data, metalsmith.metadata());
       }
-      var str = markdown.render(data.contents.toString(), env);
-      data.contents = new Buffer(str);
+
+      debug('converting file: %s', file);
```

Comment:
thank you! that should finally remove the annoying debug spam when you have non-markdown files in your filelist.